src/parse.js

```diff
@@ -602,34 +624,44 @@ function parse(source, root, options) {
         }, function parseEnumValue_line() {
             parseInlineOptions(dummy); // skip
         });
-        parent.add(token, value, dummy.comment, dummy.options);
+        parent.add(token, value, dummy.comment, dummy.options, dummy.features);
     }
 
     function parseOption(parent, token) {
-        var isCustom = skip("(", true);
-
-        /* istanbul ignore if */
-        if (!typeRefRe.test(token = next()))
-            throw illegal(token, "name");
-
-        var name = token;
-        var option = name;
-        var propName;
-
-        if (isCustom) {
-            skip(")");
-            name = "(" + name + ")";
-            option = name;
-            token = peek();
-            if (fqTypeRefRe.test(token)) {
-                propName = token.slice(1); //remove '.' before property name
-                name += token;
-                next();
+        // console.log(featuresRefRe.test(token = next()))
+        if (featuresRefRe.test(peek())) {
```

Comment:
This is kindof an unfortunate place to put the branch, since features should parse as just ordinary options.  I think a safer pattern might be to pull the features out after parsing the option by just looking for ones that start with "features."

Reply:
Similarly, we might want to keep unresolved features in the options object, since they are technically options set in the proto file.  Nobody *should* be using them and we generally prefer to hide them where possible, but there are a few cases where they're needed (e.g. producing the same .proto file you were given)